diff --git a/src/libdt.c b/src/libdt.c
index 4638678..fbbd5a2 100644
--- a/src/libdt.c
+++ b/src/libdt.c
@@ -2057,6 +2057,7 @@ static void of_scan_udev_devices(void)
 	udev_enumerate_add_match_subsystem(enumerate, "i2c");
 	udev_enumerate_add_match_subsystem(enumerate, "spi");
 	udev_enumerate_add_match_subsystem(enumerate, "mtd");
+	udev_enumerate_add_match_subsystem(enumerate, "amba");
 	udev_enumerate_scan_devices(enumerate);
 	devices = udev_enumerate_get_list_entry(enumerate);
 
