diff --git a/arch/arm/boot/dts/stm32mp151.dtsi b/arch/arm/boot/dts/stm32mp151.dtsi
index 36f38a95b4de..41ed56586322 100644
--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -87,7 +87,7 @@
 				cpu_alert1: cpu-alert1 {
 					temperature = <85000>;
 					hysteresis = <0>;
-					type = "passive";
+					type = "hot";
 				};
 
 				cpu-crit {
diff --git a/drivers/thermal/thermal_core.c b/drivers/thermal/thermal_core.c
index b71196eaf90e..8f557b1bcd2a 100644
--- a/drivers/thermal/thermal_core.c
+++ b/drivers/thermal/thermal_core.c
@@ -385,6 +385,11 @@ static void handle_critical_trips(struct thermal_zone_device *tz,
 	if (tz->ops->notify)
 		tz->ops->notify(tz, trip, trip_type);
 
+	if (trip_type == THERMAL_TRIP_HOT) {
+		dev_emerg(&tz->device,
+			  "hot temperature reached (%d C)\n",
+			  tz->temperature / 1000);
+        }
 	if (trip_type == THERMAL_TRIP_CRITICAL) {
 		dev_emerg(&tz->device,
 			  "critical temperature reached (%d C), shutting down\n",
